integration-tests/src/test/java/org/zowe/apiml/integration/ha/ChaoticTest.java

```diff
@@ -0,0 +1,118 @@
+import org.zowe.apiml.util.DiscoveryRequests;
+import org.zowe.apiml.util.categories.ChaoticHATest;
+import org.zowe.apiml.util.config.ConfigReader;
+import org.zowe.apiml.util.config.DiscoveryServiceConfiguration;
+import org.zowe.apiml.util.config.GatewayServiceConfiguration;
+import org.zowe.apiml.util.http.HttpRequestUtils;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collections;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.zowe.apiml.util.SecurityUtils.getConfiguredSslConfig;
+
+/**
+ * Verify behaviour of the application under chaotic testing
+ */
+@ChaoticHATest
+public class ChaoticTest {
+
+    private GatewayServiceConfiguration gatewayServiceConfiguration;
+    private DiscoveryServiceConfiguration discoveryServiceConfiguration;
+
+    private final String DISCOVERABLE_GREET = "/api/v1/discoverableclient/greeting";
+    private final String SHUTDOWN = "/application/shutdown";
+    final int gatewayInstances = gatewayServiceConfiguration.getInstances();
+    final int discoveryInstances = discoveryServiceConfiguration.getInstances();
+    private DiscoveryRequests discoveryRequests;
+    private String username;
+    private String password;
+    private String[] hosts;
+
+    @BeforeEach
+    void setUp() {
+        gatewayServiceConfiguration = ConfigReader.environmentConfiguration().getGatewayServiceConfiguration();
+        discoveryServiceConfiguration = ConfigReader.environmentConfiguration().getDiscoveryServiceConfiguration();
+        username = ConfigReader.environmentConfiguration().getCredentials().getUser();
+        password = ConfigReader.environmentConfiguration().getCredentials().getPassword();
+        hosts = discoveryServiceConfiguration.getHost().split(",");
+        discoveryRequests = new DiscoveryRequests(hosts[1]);
+        RestAssured.config = RestAssured.config().sslConfig(getConfiguredSslConfig());
+    }
+
+    @Nested
+    class GivenHASetUp {
+        @Nested
+        class whenOneDiscoveryServiceIsNotAvailable {
+            @Test
+            void serviceStillRegisteredToOtherDiscovery() throws URISyntaxException {
+                assumeTrue(gatewayInstances > 1 && discoveryInstances > 1);
+                String[] hosts = discoveryServiceConfiguration.getHost().split(",");
+                shutDownDiscoveryInstance(hosts[0]);
+                assertThat(discoveryRequests.isApplicationRegistered("DISCOVERABLECLIENT"), is(true));
+            }
+
+            void shutDownDiscoveryInstance(String host) throws URISyntaxException {
+                //@formatter:off
+                System.out.println(HttpRequestUtils.getUriFromDiscovery(SHUTDOWN, host));
```

Comment:
Probably either move to log or remove.

Reply:
Ah yeah sorry, it was a left over from some debugging I was doing.